src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ScheduledMetricCollectorsExecutor.java

```diff
@@ -32,11 +36,21 @@
   private boolean paEnabled = false;
   private int minTimeIntervalToSleep = Integer.MAX_VALUE;
   private Map<PerformanceAnalyzerMetricsCollector, Long> metricsCollectors;
+
+  public static final String COLLECTOR_THREAD_POOL_NAME = "pa-collectors-th";
+
+  // The next two sets helps us track collectors that are running for too long and
+  // and if they run into long collections cycles more than once we mute them.
+  private Set<PerformanceAnalyzerMetricsCollector> collectorsTakingTooLong;
+  private Set<PerformanceAnalyzerMetricsCollector> excludedCollectors;
+
   private ThreadPoolExecutor metricsCollectorsTP;
 
   public ScheduledMetricCollectorsExecutor(
       int collectorThreadCount, boolean checkFeatureDisabledFlag) {
     metricsCollectors = new HashMap<>();
+    collectorsTakingTooLong = new HashSet<>();
```

Comment:
Are these being accessed only from a single thread ? If not, we should use ConcurrentHashMap.KeySetView

Reply:
They are from a single thread.